src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clearObject = [];
+  const modifiedObject = { ...state };
+
+  actions.forEach(action => {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(modifiedObject, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete modifiedObject[key];
+        }
+        break;
+      case 'clear':
+        for (const key in modifiedObject) {
+          delete modifiedObject[key];
+        }
```

Comment:
For copy you can use just empty object
```suggestion
        modifiedObject = {};
```